src/Compilers/CSharp/Portable/Binder/Binder_Lambda.cs

```diff
@@ -113,24 +113,31 @@ internal partial class Binder
                         type = BindType(typeSyntax, diagnostics);
                         foreach (var modifier in p.Modifiers)
                         {
-                            if (modifier.Kind() == SyntaxKind.RefKeyword)
-                            {
-                                refKind = RefKind.Ref;
-                                allValue = false;
-                                break;
-                            }
-                            else if (modifier.Kind() == SyntaxKind.OutKeyword)
-                            {
-                                refKind = RefKind.Out;
-                                allValue = false;
-                                break;
-                            }
-                            else if (modifier.Kind() == SyntaxKind.ParamsKeyword)
+                            var modKind = modifier.Kind();
+
+                            switch(modifier.Kind())
                             {
-                                // This was a parse error in the native compiler; 
-                                // it is a semantic analysis error in Roslyn. See comments to
-                                // changeset 1674 for details.
-                                Error(diagnostics, ErrorCode.ERR_IllegalParams, p);
+                                case SyntaxKind.RefKeyword:
+                                    refKind = RefKind.Ref;
+                                    allValue = false;
```

Comment:
`ref readonly`   is  NYI in this change.  
 Use of `ref readonly`  was proposed just recently and will need to be added as a separate change.

There is also a change to the symbol display, with minor details still being discussed. 

Will be done in a later change.

---
In reply to: [103288914](https://github.com/dotnet/roslyn/pull/17382#discussion_r103288914) [](ancestors = 103288914)